docs/src/documentation/distributed.md

```diff
@@ -0,0 +1,118 @@
+
+To run on additional workers, a user must start additional workers in Julia and define code to run on the workers. Additional workers can be added by calling `addprocs(num_cpus)` where `num_cpus` is an integer value for the number of processors to add or start. Similarly, a user can run `rmprocs` to shutdown and remove one or more Julia processors. Once an additional worker is started, a user must also load required packages (e.g., Plasmo) on the worker they want to use. This can be done via the `@everywhere` macro, which will run the code inside the macro on every worker. For instance, the user can run the following: 
+
+```julia
+using Distributed, Plasmo, JuMP
+
+# add three processors
+println(nprocs()) # returns 1
+addprocs(3)
+println(nprocs()) # returns 4
+
+# Plasmo and JuMP are not defined on the remote workers yet until @everywhere is called
+@everywhere begin
+    using Plasmo
+    using JuMP
+end
+```
+
+To run a task on the distributed worker, a must use functions from Distributed.jl such as `remotecall` or `@spawnat`. As an example, `remotecall` will run a function on a remote worker, such as in the following case: 
+```julia
+workers = workers()
+A = rand(5000)
+
+f = remotecall(maximum, workers[1], A)
+```
+
+Here, `remotecall` runs `maximum(A)` on the first worker indexed in the worker pool. Alternatively, a user can use the `@spawnat` macro to run code such as in the following case: 
+```julia
+f = @spawnat 2 begin
+    A = rand(5000)
+    maximum(A)
+end
+```
+In this latter example, the matrix `A` is never defined on the main worker and is not accessible in the main Julia REPL because the code was run only on the remote worker. In both cases above, `f` is a `Future` object, meaning it is a reference to the task performed on the worker. Note that a user might expect this to return a Float value (the maximum value in the random vector), but the `Future` object is just a reference to what has been done on the remote worker. To get the value of the `Future` object, a user must "fetch" the value from the worker by calling `fetch(f)`, which will return the expected Float value. 
+
+
+### Cautions with Distributed Programming
+While distributed programming can be useful and accelerate task performance on some problems, there are tradeoffs. Variables, functions, or other allocated memory defined on the main processor are not shared directly with the remote workers. Thus at least two places a user may lose performance in their code is 1) making many calls to the remote worker (e.g., many `fetch` or `@spawnat` calls) and 2) passing large datastructures between the main and remote processors. A user must therefore be careful about both of these tasks. 
+
+#### Limiting calls to the remote
+As an example of the first challenge, the second function below will more efficient than the first because there is only one `remotecall` to the worker and only one `fetch` call. 
```

Comment:
Might be worth clearly saying that in this case, the cost of 2 (transferring all of A) outperforms 1 (repeated calls that transfer rows of A).